crates/astria-sequencer/src/bridge/bridge_unlock_action.rs

```diff
@@ -28,29 +30,48 @@ impl ActionHandler for BridgeUnlockAction {
         state: &S,
         from: Address,
     ) -> Result<()> {
+        // the bridge address to withdraw funds from
+        // if unset, use the tx sender's address
+        let bridge_address = self.bridge_address.unwrap_or(from);
+
         // grab the bridge account's asset
         let asset_id = state
-            .get_bridge_account_asset_id(&from)
+            .get_bridge_account_asset_id(&bridge_address)
             .await
             .context("failed to get bridge's asset id, must be a bridge account")?;
 
+        // check that the sender of this tx is the authorized withdrawer for the bridge account
+        let Some(withdrawer_address) = state
+            .get_bridge_account_withdrawer_address(&bridge_address)
+            .await
+            .context("failed to get bridge account withdrawer address")?
+        else {
+            bail!("bridge account does not have an associated withdrawer address");
+        };
```

Comment:
similar to above, what about old bridge accounts and during sync this will fail for old unlocks

Reply:
same as above, user would need to set their sudo/withdrawer addrs before using